### ISSUE TITLE
Error messages should use the same label text as the input when referring to an input

### ISSUE THREAD

**Issue by dav-idc** [alphagov/govuk-design-system#2801]
This issue is from May 2023 external accessibility audit report.
**Usability issue**

**Issue ID:**
DAC_Usability_Error_Handling_02

**URLs:**
https://govuk-frontend-pr-3503.herokuapp.com/full-page-examples/what-is-your-address

## Screen shots
<img width="529" alt="A 'What is your address?' question with a text input labelled 'Address line 1', but the error message for the input says 'Enter your building and street'." src="https://github.com/alphagov/govuk-design-system/assets/98318778/10bdbcfe-dcf2-40d0-b40b-7ed5689209df">

After committing an error with the address fields, users are presented with an error message asking them to ‘Enter your building and street’. However, the original label reads as ‘Address line 1’. It is important to provide clear and matching error handling to allow users to easily understand the error encountered and how to correct their mistakes.

## Current code ref(s)
`#main-content > div > div > form > fieldset > div:nth-child(2)`

```
<div class="govuk-form-group govuk-form-group--error">
<label class="govuk-label" for="address-line-1">
Address line 1
</label>
<p id="address-line-1-error" class="govuk-error-message">
<span class="govuk-visually-hidden">Error:</span> Enter your building and street </p>
<input class="govuk-input govuk-input--error" id="address-line-1" name="address-line-1" type="text" aria-describedby="address-line-1-error" autocomplete="address-line1">
</div>
```

### Screen reader user
> “The first input field on the page announced to me as ‘Address line 1’ which was clear to me both in and out of context. However, when I committed an error the error skip link and inline error validation message corresponding to the input field read to me as ‘Enter your building and street’. This presented different content and may for some screen reader users cause confusion as if an error has already been committed the user may not be completely aware of what must be entered. Updating the error skip link and inline error validation message to match the wording used in the input field label will ensure that all users are aware of what should be entered.”

## Solution
Consider changing either the label or the error message, so that they both match and to provide all users with a clear and concise error handling and labelling for the input fields.

**Comment by CharlotteDowns** [comment]
@davidc-gds quick question do we think this is already covered in the error message guidance

> Use the question or form label in the error to provide context. For example, ‘Enter how many hours you work a week’ for ‘How many hours do you work a week?’

Obviously this was my poor interpretation of that line that led to this issue 🙈.

**Comment by davidc-gds** [comment]
Hi @CharlotteDowns sorry for not responding sooner my inbox has been flooded this week.

I don't think the current guidance is enough to prevent this issue, as like you pointed out, we're not even able to successfully interpret it ourselves.

**Comment by CharlotteDowns** [comment]
@davidc-gds I wonder if there's more nuance to this and we need to double check the guidance between the addresses pattern and the error message guidance. We refer to Address line 1 in the addresses pattern but that makes it hard to fit our error message guidance "You must enter Address line 1"???

**Comment by davidc-gds** [comment]
@CharlotteDowns I'm guessing for cognitive considerations the idea is "if an input is labelled 'address line 1' then in an error message, it should be theoretically be good enough to refer to the input as 'address line 1'.
So basically if the input's label doesn't make sense in the context of an error message, then maybe it's the wrong input label.

**Comment by davidc-gds** [comment]
A conversation / work session will be the next step for this piece.
@CharlotteDowns has booked a chat for Friday 25 August for Charlotte, Calvin and me.

**Comment by CharlotteDowns** [comment]
We had a chat, these are the proposed next steps:

1. On the addresses pattern page include two error previews:
{preview code example – all unfilled required fields} just the label repeated.
If a postcode is not a real postcode
{preview code example – other example}

2. New heading above 'Be clear and concise' on the error message component page
'Use the inputs label as the error message' and update the legend preview code example to 'Select your nationality' adding paragraph copy on referring to the legend when there is one question 'In most cases just repeat the key part of the question'. Series of 'if' statements maybe.

**Comment by dav-idc** [comment]
With those 2 GitHub issues created, does that close out this issue? We can continue work on the 2 new issues.

**Comment by dav-idc** [comment]
I've added those 2 issues to the tasklist for the epic ticket above this one. Will close this ticket now as done, and we can work on the one uncompleted ticket.
-  https://github.com/alphagov/govuk-design-system/issues/3097
